overlord/hookstate/ctlcmd/set.go

```diff
@@ -19,15 +19,29 @@
 
 package ctlcmd
 
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+)
+
 type setCommand struct {
 	baseCommand
 }
 
+var shortSetHelp = i18n.G("Set snap configuration")
+var longSetHelp = i18n.G(`
+The set command is currently only a placeholder.`)
+
 func init() {
-	addCommand("set", func() command { return &setCommand{} })
+	addCommand("set", shortSetHelp, longSetHelp, func() command { return &setCommand{} })
 }
 
 func (s *setCommand) Execute(args []string) error {
+	if s.context() == nil {
```

Comment:
yes, it seems to me we could do something in Run instead of delaying the check here. The command could have a way to declare if they need context (afaiu they actually all need it atm) and we could pass a dummy command object to AddCommand that fails like this instead of the real object command if the context is nil